### DISCUSSION TITLE
Use once URL for photo/album sharing

### DISCUSSION THREAD

**Discussion by lxs** [tripupapp/tripup-server#1] [Ideas]
**For Your Eyes Only, but Once Only**

**Gist:** URL can be opened once only, and then becomes invalid. 

**Alternatively:** endpoint can be opened a limited number of times before becoming invalid (example: The Financial Times allows subscribed members to share an article via a URL, yet this can only be opened for free 3 times -- and then they pull up the newspaper's "pay-wall"). This would serve to limit the scope of availability.

**Why?** if an intermediary were to evaluate what is at the endpoint then the recipient is denied access, and would suspect a compromise has occurred.

**Why not?** not helpful when sharing URL to group which would be anticipated to have the endpoint opened multiple times (e.g. sharing photo album with family/friends).

**Why not?** likely to be incompatible with antivirus or enterprise tools that serve to detect malicious endpoints (i.e. tools that evaluate the endpoint to determine if safe before then allowing this to be accessed/opened)

**Comment by vin047** [comment]
Interesting idea. I'll have to look into this further to see how it could be implemented. What sort of use cases do you see for such a feature?

**Reply by lxs** [reply]
Was shared as ideation...

Mention the newspaper pay-wall as this was a scenario that came to mind -- namely to offer a free snippet view of image. This might serve to demonstrate a style or skill without wanting this to leak in broader domain. To entice a recipient to commit to negotiate initial copyright/licensing terms for the photos/digital media.

This is always going to be of questionable merit for digital content -- as tricky to manage control of a digital asset once it has been "displayed" to a recipient -- unless you have control of their system(s) to avoid a screenshot or other recording effort.
